gramps/gen/filters/rules/family/_fatherhasidof.py

```diff
@@ -48,5 +56,10 @@ class FatherHasIdOf(RegExpIdBase):
     name = _("Families having father with Id containing <text>")
     description = _("Matches families whose father has a specified " "Gramps ID")
     category = _("Father filters")
-    base_class = RegExpIdBase
-    apply = father_base
+
+    def apply_to_one(self, db: Database, family: Family) -> bool:  # type: ignore[override]
+        father_handle = family.father_handle
+        if father_handle:
+            father = db.get_person_from_handle(father_handle)
+            return super().apply_to_one(db, father) if father else False
```

Comment:
```suggestion
            if father:
                return super().apply_to_one(db, father)
```
I think this is slightly clearer maybe?